client/src/main/kotlin/tech/figure/validationoracle/client/client/impl/DefaultVOExecutor.kt

```diff
@@ -10,48 +10,83 @@ import io.provenance.client.grpc.Signer
-import tech.figure.validationoracle.client.domain.execute.base.ContractExecute
+import tech.figure.validationoracle.client.domain.execute.EntityCreationRequest
+import tech.figure.validationoracle.client.domain.execute.EntityUpdateRequest
+import tech.figure.validationoracle.client.domain.execute.ValidationDefinitionCreationRequest
+import tech.figure.validationoracle.client.domain.execute.ValidationDefinitionDeletionRequest
+import tech.figure.validationoracle.client.domain.execute.ValidationDefinitionUpdateRequest
+import tech.figure.validationoracle.client.domain.execute.ValidationRequest
+import tech.figure.validationoracle.client.domain.execute.ValidationRequestDeletion
+import tech.figure.validationoracle.client.domain.execute.ValidationRequestUpdate
+import tech.figure.validationoracle.client.domain.execute.base.ContractExecuteInput
 
 /**
- * The default implementation of an [VOExecutor].  Provides all the standard functionality to use an [VOClient][tech.figure.validationoracle.client.client.base.VOClient] if an
- * override for business logic is not necessary.
+ * The default implementation of an [VOExecutor]. Provides all the standard functionality to use a
+ * [VOClient][tech.figure.validationoracle.client.client.base.VOClient]
+ * if an override for business logic is not necessary.
  */
 class DefaultVOExecutor(
     private val objectMapper: ObjectMapper,
     private val pbClient: PbClient,
     private val querier: VOQuerier,
 ) : VOExecutor {
-    private fun generateAddValidationDefinitionMsg(
-        execute: AddValidationDefinitionExecute,
-        signerAddress: String,
-    ): MsgExecuteContract = generateMsg(execute, signerAddress)
 
-    override fun addValidationDefinition(
-        execute: AddValidationDefinitionExecute,
+    override fun createValidationDefinition(
+        request: ValidationDefinitionCreationRequest,
         signer: Signer,
-        options: tech.figure.validationoracle.client.client.base.BroadcastOptions,
-    ): BroadcastTxResponse = doExecute(generateAddValidationDefinitionMsg(execute, signer.address()), signer, options)
+        options: BroadcastOptions,
+    ): BroadcastTxResponse = doExecute(generateMsg(request, signer.address()), signer, options)
 
-    private fun generateAddValidationDefinitionMsg(
-        execute: RequestValidationExecute,
-        signerAddress: String,
-    ): MsgExecuteContract = generateMsg(execute, signerAddress)
+    override fun updateValidationDefinition(
+        request: ValidationDefinitionUpdateRequest,
+        signer: Signer,
+        options: BroadcastOptions,
+    ): BroadcastTxResponse = doExecute(generateMsg(request, signer.address()), signer, options)
+
+    override fun deleteValidationDefinition(
+        request: ValidationDefinitionDeletionRequest,
+        signer: Signer,
+        options: BroadcastOptions,
+    ): BroadcastTxResponse = doExecute(generateMsg(request, signer.address()), signer, options)
```

Comment:
Do these all do the same thing?

Reply:
Yes

Reply:
So, why the need for separately named methods?

Reply:
I just realized these all return `BroadcastTxResponse`s when the ideal interface would be requiring them to return domain objects — ones which I already started defining in this PR 🤦🏽 If you'd agree that this would have more value if it did, then allow me to address that in a follow-up PR.

Reply:
Okay, now that it's no longer Friday I seem to be seeing this clearly enough to recognize that it's _intentionally_ not returning domain objects because it simply does not make sense to. Querying a provwasm contract returns a JSON response completely customized by the contract code, but executing a provwasm contract returns a Provenance transaction response. This response contains details both relating to what the contract did and also additional attributes determined by the contract's logic, but in the end is ultimately a reflection of activity done on chain, and thus probably should continue to return deserialized `BroadcastTxResponse` protobufs. Let me know if you have further thoughts on that.